testing/internal/e2e/tests/base/target_pagination_test.go

```diff
@@ -0,0 +1,181 @@
+)
+
+// TestCliPaginateTargets asserts that the CLI automatically paginates to retrieve
+// all targets in a single invocation.
+func TestCliPaginateTargets(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId)
+
+	// Create enough targets to overflow a single page.
+	// Use the API to make creation faster.
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	tClient := targets.NewClient(client)
+	targetPort, err := strconv.ParseInt(c.TargetPort, 10, 32)
+	require.NoError(t, err)
+	for i := 0; i < globals.DefaultMaxPageSize+1; i++ {
+		_, err := tClient.Create(ctx, "tcp", newProjectId,
+			targets.WithName("test-target-"+strconv.Itoa(i)),
+			targets.WithTcpTargetDefaultPort(uint32(targetPort)),
+			targets.WithAddress(c.TargetAddress),
+		)
+		require.NoError(t, err)
+	}
+
+	// List targets recursively
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"-scope-id", newProjectId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var initialTargets targets.TargetListResult
+	err = json.Unmarshal(output.Stdout, &initialTargets)
+	require.NoError(t, err)
+
+	require.Len(t, initialTargets.Items, globals.DefaultMaxPageSize+1)
+	// Note that none of these are returned to the CLI for now.
+	assert.Empty(t, initialTargets.ResponseType)
+	assert.Empty(t, initialTargets.RemovedIds)
+	assert.Empty(t, initialTargets.RefreshToken)
+
+	// Create a new target and destroy one of the other targets
+	newTargetId := boundary.CreateNewTargetCli(t, ctx, newProjectId, c.TargetPort, target.WithAddress(c.TargetAddress))
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "delete",
+			"-id", initialTargets.Items[0].Id,
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// List again, should have the new target but not the deleted target
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"targets", "list",
+			"-scope-id", newProjectId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var newTargets targets.TargetListResult
+	err = json.Unmarshal(output.Stdout, &newTargets)
+	require.NoError(t, err)
+
+	require.Len(t, initialTargets.Items, globals.DefaultMaxPageSize+1)
+	assert.Equal(t, newTargetId, newTargets.Items[len(newTargets.Items)-1].Id)
+	assert.Empty(t, initialTargets.ResponseType)
+	assert.Empty(t, initialTargets.RemovedIds)
+	assert.Empty(t, initialTargets.RefreshToken)
```

Comment:
Was this meant to validate `newTargets`?

```
require.Len(t, newTargets.Items, globals.DefaultMaxPageSize)
assert.Equal(t, newTargetId, newTargets.Items[len(newTargets.Items)-1].Id)
assert.Empty(t, newTargets.ResponseType)
assert.Empty(t, newTargets.RemovedIds)
assert.Empty(t, newTargets.RefreshToken)
```

Reply:
Doh, good shout